data/pigui/modules/new-game-window/ship.lua

```diff
@@ -546,14 +546,14 @@ function ShipEquip:getHyperDriveClass()
 	return 0
 end
 
+-- returns 0..3 representing default to naval
 function ShipEquip:getThrusterUpgradeLevel()
```

Comment:
I only uncommented what was there already.. but I can revert this. Looks like it's not used anywhere at present anyway. 

Trying to debug why the ship could still fly even when I changed all the formula so that missing thrusters gave a multiplier of 0 instead of 1 - turned out that this code is completely unused, and most of the LUA-side is only for displaying data; the real magic is in the C++..

Reply:
Yep - if you've not already found it, this is handled in `Ship::UpdateEquipStats()` and `Propulsion:SetThrustPowerMult()` in C++.